.github/workflows/binary-publish.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish binary
+on:
+  workflow_dispatch:
+
+jobs:
+  SetEnv:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get version string
+        id: version
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags/') }}; then
+            echo "version=$(git describe --exact-match --tags $(git rev-parse HEAD))" | tee -a $GITHUB_OUTPUT
+          else
+            echo "version=$(echo unreleased-$(git rev-parse --short HEAD))" | tee -a $GITHUB_OUTPUT
```

Comment:
Few comments:

1. For unreleased versions, should we also attach branch name `--abbrev-ref` along with commit id? Probably for easier bookkeeping. 
2. Not related to this change, but just curious - When we share unreleased binaries with customers, do we share any terms & conditions with them?

Reply:
> For unreleased versions, should we also attach branch name --abbrev-ref along with commit id? Probably for easier bookkeeping.

- In most cases the unknown version will be released from `main` and so won't really have a useful branch name.
- We have been using this version format for quite some time and it would be better to stick to it
- The users have no real use for the branch name, and we can always just lookup the branch name from commit

> Not related to this change, but just curious - When we share unreleased binaries with customers, do we share any terms & conditions with them?

Apart from the fact that we highlight that it is a development build and not a release, nothing else.

Reply:
Ah okay, didn't know that `main` would be the source. Thanks for answering my questions!